src/sentry/api/urls.py

```diff
@@ -1573,6 +1578,21 @@
                     ProjectMemberIndexEndpoint.as_view(),
                     name="sentry-api-0-project-member-index",
                 ),
+                url(
+                    r"^(?P<organization_slug>[^/]+)/(?P<project_slug>[^/]+)/metrics/$",
+                    ProjectMetricsEndpoint.as_view(),
+                    name="sentry-api-0-project-metrics-index",
+                ),
+                url(
+                    r"^(?P<organization_slug>[^/]+)/(?P<project_slug>[^/]+)/metrics/data/$",
+                    ProjectMetricsDataEndpoint.as_view(),
+                    name="sentry-api-0-project-metrics-data",
+                ),
+                url(
+                    r"^(?P<organization_slug>[^/]+)/(?P<project_slug>[^/]+)/metrics/tags/(?P<metric_name>[^/]+)/(?P<tag_name>[^/]+)/$",
```

Comment:
First off: This API makes sense in my opinion. It is not REST-like however. Taking a stab at a REST-version of it, you tell me if you like this more :)

```
GET /metrics/                       # list all metrics, low detail (no tags)
GET /metrics/<metric>/              # list details of metric incl tags, no tag values. query param for tag values?
GET /metrics/<metric>/tags/         # list only tags of metric, no values. query param for tag values?
GET /metrics/<metric>/tags/<tag>/   # list tag values for this tag

# one of:
POST /metrics-data/       # queries, using POST because of request body
GET /metrics-data/?...    # queries, using verbose query params
```

Note we went down the POST route for sessions, so we should probably stick to post.

Reply:
I like the REST idea. @priscilawebdev what do you think?
The sessions API is `GET` only, as far as I can see.

Reply:
I like the REST idea :)

Reply:
As Jan pointed out, we will need to query tags across metrics. The URL schema would then look something like

```
/metrics/metrics/
...
/metrics/tags/
...
/metrics/data/
```

Would be nice if we had different words for the concept of metrics and the individual metric instances.

Reply:
You could go with either `/metrics/meta/` or `/metrics/schema/` to make it less odd.

Reply:
I've created a separate PR for this thread: https://github.com/getsentry/sentry/pull/25395